pkg/ui/src/Forms/BookDetailForm.svelte

```diff
@@ -12,23 +12,103 @@
 	export let book: BookEntry;
 	export let publisherList: string[] = [];
 	export let onSubmit: (values: BookEntry) => void = () => {};
+	export let editMode: boolean = false;
+	export let openEditMode = () => {};
 	export let onCancel = () => {};
+	export let onISBNField: (isbn: string) => Promise<BookEntry[]> | undefined = () => undefined;
 
-	const { form } = createForm({
+	interface BookCache {
+		[key: string]: BookEntry;
+	}
+
+	// book store which can  be set
+	const bookCache: BookCache = {};
+
+	// reactive declaration for instant error feedback (felte returns stores on blur i think)
+	$: isbnExistsError = "";
+
+	const { form, data, setFields, setTouched, interacted } = createForm({
 		initialValues: book,
 		onSubmit: (values) => {
 			onSubmit(values);
+		},
+		debounced: {
+			// defaults to 300
+
+			timeout: 1000,
+			validate: async (values) => {
+				const errors: {
+					isbn?: string;
+					title?: string;
+					price?: string;
+					year?: string;
+					authors?: string;
+					publisher?: string;
+					editedBy?: string;
+					outOfPrint?: string;
+				} = { isbn: isbnExistsError };
+				const isbnError = "ISBN already exists, would you like to Edit it?";
+
+				if (values.isbn.trim() === "" || $interacted !== "isbn") return errors;
+
+				// reset isbn error if isbn field interacted with
+				isbnExistsError = "";
+
+				if (bookCache[values.isbn]) {
+					errors.isbn = isbnError;
+					isbnExistsError = isbnError;
+				} else {
```

Comment:
yeah, the book is still cached, but this would only be checked for if the isbn field was interacted with which it won't be when the user clicks edit/autofill